File: solutions/go/clock/2/clock.go
package clock

import "fmt"

//Clock contains date information with only hours and minutes
type Clock struct {
	hour, minute int
}

const (
	minutesInHour = 60
	hoursInDay    = 24
)

//New creates Clock instance
func New(hour int, minute int) Clock {
	hour += minute / minutesInHour
	minute %= minutesInHour
	if minute < 0 {
		minute += minutesInHour
		hour--
	}
	hour %= hoursInDay
	if hour < 0 {
		hour += hoursInDay
	}
	return Clock{hour: hour, minute: minute}
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

//Add return new Clock with addition minutes to the current
func (c Clock) Add(minutes int) Clock {
	return New(c.hour, c.minute+minutes)
}

//Subtract return new Clock with subtraction minutes from the current
func (c Clock) Subtract(minutes int) Clock {
	return New(c.hour, c.minute-minutes)
}

File: solutions/swift/clock/1/Sources/Clock/Clock.swift
import Foundation

struct Clock {
    private static let minutesInHour = 60
    private static let hoursInDay = 24

    var hours: Int
    var minutes: Int

    init(hours: Int = 0, minutes: Int = 0) {
        var minutes = minutes
        var hours = hours + minutes / Self.minutesInHour
        minutes %= Self.minutesInHour
        if minutes < 0 {
            minutes += Self.minutesInHour
            hours -= 1
        }
        hours %= Self.hoursInDay
        if hours < 0 {
            hours += Self.hoursInDay
        }

        self.hours = hours
        self.minutes = minutes
    }

    func add(minutes: Int) -> Clock {
        Clock(hours: hours, minutes: self.minutes + minutes)
    }

    func subtract(minutes: Int) -> Clock {
        Clock(hours: hours, minutes: self.minutes - minutes)
    }
}

extension Clock: Equatable {}

extension Clock: CustomStringConvertible {
    var description: String {
        String(format: "%02ld:%02ld", hours, minutes)
    }
}
